diff --git a/setup.py b/setup.py
index a15ef2b..5eb720e 100644
--- a/setup.py
+++ b/setup.py
@@ -177,6 +177,12 @@ def main():
         ver_dic,
     )
 
+    # Auto-generate local version from CUDA version.
+    import os
+    cuda_version = os.environ.get("MANYLINUX_CUDA_VERSION")
+    if cuda_version:
+        ver_dic["VERSION_TEXT"] = f"{ver_dic['VERSION_TEXT']}+cu.{cuda_version}"
+
     import sys
 
     setup(
